### PR TITLE
Update gitignore with the latest VisualStudio entry from github/gitignore

### PR DESCRIPTION
**Pull request by haacked**
I also updated the python entries. They seemed to be catching too much and I assume those were meant to be directories.

### CODE REVIEW

**Comment by haacked** [comment]
@xpaulbettsx, @niik, or @half-ogre. Would love another set of eyes on this. Should only take a second. :)

File: etc/gitignore.suggested [hunk 1]
```diff
@@ -59,12 +69,14 @@ local.properties
...
 *.tlh
 *.tmp
+*.tmp_proj
+*.log
 *.vspscc
+*.vssscc
 .builds
-*.dotCover
-
-## TODO: If you have NuGet Package Restore enabled, uncomment this
```
Comment (half-ogre) [inline etc/gitignore.suggested:66]: Why take this out? I guess if you';re using package restore you'll know to add the packages folder?

**Comment by half-ogre** [comment]
Looks fine to me. One comment about the NuGet packages folder, but nothing to keep from :shipit:.

File: etc/gitignore.suggested [hunk 2]
```diff
@@ -94,38 +122,66 @@ DocProject/Help/Html2
...
 # Click-Once directory
-publish
+publish/
+
+# Publish Web Output
+*.Publish.xml
+*.pubxml
+
+# NuGet Packages Directory
+## TODO: If you have NuGet Package Restore enabled, uncomment the next line
```
Comment (haacked) [inline etc/gitignore.suggested:132]: @half-ogre: It was moved down here!

File: etc/gitignore.suggested (hunk 1 above)
Comment (haacked) [inline etc/gitignore.suggested:66]: It was moved down a bit. :)

**Pull request merged**